waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -396,7 +396,7 @@ proc init*(ws: WakuStore) {.raises: [Defect, Exception]} =
 
 
 proc init*(T: type WakuStore, peerManager: PeerManager, rng: ref BrHmacDrbgContext,
-                   store: MessageStore = nil, wakuSwap: WakuSwap = nil, persistMessages = true): T {.raises: [Defect, Exception]} =
```

Comment:
Why is this removed?

Reply:
In general, raising the non-specific `Exception` is bad practice. Here it was done because the underlying modules weren't explicit about what they're raising. Now that the underlying modules have better error-handling, the top-level `{.push raises: [Defect].}` is enough for this function.